v1/plugin/stream_collector_proxy.go

```diff
@@ -123,33 +134,46 @@ func (p *StreamProxy) metricSend(ch chan []Metric, stream rpc.StreamCollector_St
 			// send metrics if maxCollectDuration is reached
 			sendReply(metrics, stream)
 			metrics = []*rpc.Metric{}
-
+		case <-stream.Context().Done():
+			return
 		}
 	}
 }
 
 func (p *StreamProxy) streamRecv(ch chan []Metric, stream rpc.StreamCollector_StreamMetricsServer) {
+	log.WithFields(
+		log.Fields{
+			"_block": "streamRecv",
+		},
+	).Debug("receiving metrics")
 	for {
-		s, err := stream.Recv()
-		if err != nil {
-			fmt.Println(err)
-			continue
-		}
-		if s != nil {
-			if s.MaxMetricsBuffer > 0 {
-				p.setMaxMetricsBuffer(s.MaxMetricsBuffer)
-			}
-			if s.MaxCollectDuration > 0 {
-				p.setMaxCollectDuration(time.Duration(s.MaxCollectDuration))
+		select {
+		case <-stream.Context().Done():
+			close(ch)
+			return
+		default:
+
+			s, err := stream.Recv()
+			if err != nil {
+				log.Debug(err)
```

Comment:
@jcooklin , could you confirm this log should stay on debug level, or maybe "error" would be more appropriate here?  What do you think about keeping the same pattern and defining log with the field "_block".